openmc/tallies.py

```diff
@@ -1482,19 +1482,31 @@ def _outer_product(self, other, binary_op):
             new_name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
             new_tally.name = new_name
 
+        # Create copies of self and other tallies to rearrange for tally
+        # arithmetic
+        self_copy = copy.deepcopy(self)
+        other_copy = copy.deepcopy(other)
+
         # Find any shared filters between the two tallies
-        self_filters = set(self.filters)
-        other_filters = set(other.filters)
-        filter_intersect = self_filters.intersection(other_filters)
+        filter_intersect = []
+        for filter in self_copy.filters:
+            if filter in other_copy.filters:
+                filter_intersect.append(filter)
```

Comment:
@wbinventor you were right! It was the ordering of `filter_intersect` that was leading to the inconsistency between Python 2 and 3. Converting the filters lists to sets resulted in their order being changed in Python 3. Finding the intersection manually avoids this issue.